rfcs/syntax-lambda-function-expressions.md

```diff
@@ -0,0 +1,104 @@
+# Lambda function expressions 
+
+## Summary
+
+Introduce a way to define anonymous functions using a terse syntax.
+
+## Motivation
+
+Current function syntax takes up a lot of space when used for short anonymous that are passed in function calls.
+
+By introducing a shorter syntax, we make code with heavy use of anonymous functions easier to write and arguably, easier to read as well.
+
+Existing syntax:
+```lua
+local b = filter(map(c, function(a) return a * 2 end), function(a) return a > 0 end)
+local b = filter(c, function(a) return a.good end)
+local a = sort(c, function(a, b) return a.x > b.x end)
+```
+
+This proposal:
+```lua
+local b = filter(map(c, a -> a * 2), a -> a > 0)
```

Comment:
Missing introduction token here and below

Reply:
This is the special one-argument syntax.

Reply:
![image](https://user-images.githubusercontent.com/1106629/121567715-c11b3400-c9d3-11eb-960c-5398efdb7a9e.png)

Reply:
That seems a lot like making a third syntax for functions... Like sometimes you need `do`, sometimes you don't, seems to me like it'd be more consistent to standardize on `do(a) -> a * 2`.